numpy/lib/nanfunctions.py

```diff
@@ -579,10 +604,15 @@ def nansum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
 
     """
     a, mask = _replace_nan(a, 0)
-    return np.sum(a, axis=axis, dtype=dtype, out=out, keepdims=keepdims)
+
+    if initializer is not np._NoValue:
+        initializer, mask = _replace_nan(initializer, 0)
```

Comment:
Since the mask is not used, maybe `initializer, _  = _replace_nan(initializer, 0)`

Actually, do we want to bother to replace it at all? Why let the user silently pass in nonsense?

Reply:
The reasoning here is; since these are `nan`-skipping reductions, they should skip the `nan` initializer as well. Otherwise, they won't have the effect of being nan-skipping, would they?

Reply:
But why apply that to the initializer? I just don't see any use case for the increased complexity, and I do see a case for giving an error/warning when input is unexpected! In other words, I think most if not all cases where `initializer=np.nan` are going to be erroneous, and this might as well lead to an error.

Reply:
I'm not sure, but I do agree this is worth discussing. Maybe we can defer to someone else for this case?

Reply:
@eric-wieser - what do you think? Does it make sense to replace `nan` in the initlializer with `0` here, or should we just let things fail for such input?

Reply:
We should choose such that

```
func(arr_1, initializer=func(arr_2, initializer=x))
```
is always treated the same as

```
func(concatenate((arr_1, arr_2)), initializer=x)
```

I think this is maintained in either case?

Reply:
For @mhvk's suggestion, it isn't maintained for `initializer=nan`

Reply:
Do you mean for `x == nan`? Can you show the code that fails that test?

Reply:
If in both cases one gets an exception, that would be fine too!